src/components/rangeslider/draw.js

```diff
@@ -279,17 +293,102 @@ function drawBg(rangeSlider, gd, axisOpts, opts) {
     });
 }
 
-function drawRangePlot(rangeSlider, gd, axisOpts, opts) {
-    var rangePlots = rangePlot(gd, opts._width, opts._height);
+function addClipPath(rangeSlider, gd, axisOpts, opts) {
+    var fullLayout = gd._fullLayout;
 
-    var gRangePlot = rangeSlider.selectAll('g.' + constants.rangePlotClassName)
+    var clipPath = fullLayout._topdefs.selectAll('#' + opts._clipId)
         .data([0]);
 
-    gRangePlot.enter().append('g')
-        .classed(constants.rangePlotClassName, true);
+    clipPath.enter().append('clipPath')
+        .attr('id', opts._clipId)
+        .append('rect')
+        .attr({ x: 0, y: 0 });
+
+    clipPath.select('rect').attr({
+        width: opts._width,
+        height: opts._height
+    });
+}
+
+function drawRangePlot(rangeSlider, gd, axisOpts, opts) {
```

Comment:
So, the algorithm is as follows:

1. find all the subplots spanned by the axis where the range slider reside
1. for each subplot, mock a `plotinfo` object (which are essentially the cartesian subplot state objects) using range slider options to find the width and height
1. find all traces in this subplot
1. Call `Cartesian.rangePlot` which creates all the required layers and plots / styles the given `plotinfo` subplot
1. clip the subplot